### PR TITLE
Added ticket replies (email) and notes to linked issue as a comment

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/163697/wrike-new-features

### CODE REVIEW

File: src/hooks/useReplyBox.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+  }
+
+  if (Array.isArray(taskIds) && !taskIds.length) {
+    return client.deregisterTargetAction(`${APP_PREFIX}ReplyBoxEmailAdditions`);
+  }
+
+  return Promise
+    .all(taskIds.map((taskId: ITaskFromList["id"]) => {
+      return client.getState<{ selected: boolean }>(emailKey(ticketId, taskId))
+    }))
```
Comment (HappyPaul55) [inline src/hooks/useReplyBox.tsx:92]: Could this be expensive if we get a lot of taskIds? Instead, can we wait for one to finish before getting the next?

```
const results = [];
for (const taskId of taskIds) {
    results.push(await client.getState<{ selected: boolean }>(emailKey(ticketId, taskId)))
}

// extra stuff

return results;
```

File: src/hooks/useReplyBox.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+const useReplyBox = () => useContext<ReturnUseReplyBox>(ReplyBoxContext);
+
+const ReplyBoxProvider: FC<PropsWithChildren> = ({ children }) => {
+  const { context } = useDeskproLatestAppContext();
+  const { client } = useDeskproAppClient();
+  const { tasks } = useLinkedTasks();
+  const tasksMap = useMemo(() => tasks.reduce<Record<ITaskFromList["id"], ITaskFromList>>((acc, task) => {
+      acc[task.id] = task;
+      return acc;
+    }, {}), [tasks]);
```
Comment (HappyPaul55) [inline src/hooks/useReplyBox.tsx:123]: No need for a memo here, creating a map out of an array of tasks is cheap for the CPU to do - even if large.

**Review by HappyPaul55 — CHANGES_REQUESTED** [review]

File: src/hooks/useReplyBox.tsx (hunk 2 above)
Comment (zpawn) [reply src/hooks/useReplyBox.tsx:123]: fixed

File: src/hooks/useReplyBox.tsx (hunk 1 above)
Comment (zpawn) [reply src/hooks/useReplyBox.tsx:92]: Honestly, I doubt that many tasks can be linked to a ticket. We can optimize this later if needed. Or we can do it now if you think it’s important.

File: src/components/common/Button/Button.tsx [hunk 3]
```diff
@@ -3,18 +3,21 @@ import styled from "styled-components";
...
 
-export const ButtonAsLink = styled.button<{ color?: keyof ThemeColors }>`
+export const ButtonAsLink = styled.button<ButtonAsLinkProps>`
   display: inline;
   background: none;
   border: none;
   padding: 0;
   font: inherit;
   cursor: pointer;
+  text-align: ${({ align }) => align || "left" };
```
Comment (HappyPaul55) [inline src/components/common/Button/Button.tsx:20]: minor: `?? "left"` is probably more accurate.

File: src/hooks/useReplyBox.tsx (hunk 1 above)
Comment (HappyPaul55) [reply src/hooks/useReplyBox.tsx:92]: Fair enough, happy to leave as is for now, but maybe leave a tech debt ticket or note somewhere saying that this should be revisited if clients often link many-many tickets.

**Review by HappyPaul55 — APPROVED** [review]

File: src/components/common/Button/Button.tsx (hunk 3 above)
Comment (zpawn) [reply src/components/common/Button/Button.tsx:20]: fixed

File: src/hooks/useReplyBox.tsx (hunk 1 above)
Comment (zpawn) [reply src/hooks/useReplyBox.tsx:92]: added comment in code

**Pull request merged**